core/src/services/s3/backend.rs

```diff
@@ -1457,7 +1457,12 @@ impl S3Backend {
         let source = format!("{}/{}", self.bucket, percent_encode_path(&from));
         let target = format!("{}/{}", self.endpoint, percent_encode_path(&to));
 
-        let mut req = Request::put(&target)
+        let mut req = Request::put(&target);
+
+        // Set SSE headers.
+        req = self.insert_sse_headers(req, true);
```

Comment:
I think we also need to set `X_AMZ_COPY_SOURCE_SSE_XXX`

Reply:
This doesn't seem to need to be set up. I don't see it mentioned in the documentation either. It's actually a put_object with an extra header `copy_source`.

Reply:
> It's actually a put_object with an extra header `copy_source`.

If bucket has been configured with SSE, all written object will be encrypted. To copy from it, we need to set `x-amz-copy-source-server-side-encryption-customer-algorithm` and related headers.

Reply:
I see. Let's have a try.